test/unit/source-change.test.js

```diff
@@ -0,0 +1,341 @@
+/* eslint-env qunit */
+import videojs from '../../src/js/video.js';
+import document from 'global/document';
+import window from 'global/window';
+
+const testSrc = {
+  src: 'http://vjs.zencdn.net/v/oceans.mp4',
+  type: 'video/mp4'
+};
+
+QUnit.module('sourcechange', function() {
+  QUnit.module('videojs sources', {
```

Comment:
I think this is all the ways to set sources, feel free to tell me I need more here though

Reply:
Basically 2 main methods: via a src attribute or source element. And these have corresponding vjs options like the `sources` option and also just calling .src() on the player. So, guess it makes 4 total ways of doing it.